### PR TITLE
Distribute separate and modified run.sh for docker/pypi/reqmgr2ms-unerged

### PR DESCRIPTION
**Pull request by todor-ivanov**
Fixes: https://github.com/dmwm/WMCore/issues/11637 
This is a fix for the broken `run.sh` for `pypi/reqmgr2ms-unmerged` images inherited from the `pypi/dmwm-base` image.

The so distributed file is and exact copy of the original `run.sh`, with only two modifications:
* Adding an option to accept the `RSEExpr` as external parameter 
* Exporting the `X509_USER_PROXY` environment variable

Here follows the actual diff between the two files
```
--- docker/pypi/dmwm-base/run.sh        2023-06-29 12:04:20.000000001 +0200
+++ docker/pypi/reqmgr2ms-unmerged/run.sh       2023-07-13 18:25:36.000000001 +0200
@@ -1,5 +1,59 @@
 #!/bin/bash
-# script to start ReqMgr2
+# script to start MSUnmerged
+
+
+### Usage: Usage: run.sh [-e 'RSE Expression']
+### Usage:
+### Usage:   -e  <RSE Expression>   A Rucio RSE Expression to be passed to the service configuration
+### Usage:                          before service start
+### Usage:   -h <help>              Provides help to the current script
+### Usage:
+### Usage: Example: ./run.sh -e 'rse_type=DISK&country=US&tier=3&cms_type=real'
+### Usage: Example: ./run.sh
+### Usage:
+
+FULL_SCRIPT_PATH="$(realpath "${0}")"
+
+usage()
+{
+    echo -e $*
+    grep '^### Usage:' < $FULL_SCRIPT_PATH
+    exit 1
+}
+
+help()
+{
+    echo -e $*
+    grep '^### Usage:' < $FULL_SCRIPT_PATH Reviewers 
+    exit 0
+}
+
+### Searching for the mandatory and optional arguments:
+# export OPTIND=1
+while getopts ":e:h" opt; do
+    case ${opt} in
+        e)
+            rseExpr=${OPTARG}
+            # escaping all possible special characters in the RSE expression:
+            # NOTE: One should always start escaping the es Reviewers caping chars first
+            rseExpr=${rseExpr//\\/\\\\}
+            rseExpr=${rseExpr//&/\\&}
+            rseExpr=${rseExpr//|/\\|}
+            ;;
+        h)
+            help
+            ;;
+        \? )
+            msg="Invalid Option: -$OPTARG"
+            usage "$msg"
+            ;;
+        : )
+            msg="Invalid Option: -$OPTARG requires an argument"
+            usage "$msg"
+            ;;
+    esac
+done
+
 
 srv=`echo $USER | sed -e "s,_,,g"`
 STATEDIR=/data/srv/state/$srv
@@ -23,7 +77,7 @@
 export WMCORE_CACHE_DIR=$STATEDIR
 
 # overwrite host PEM files in /data/srv area by the robot certificate
-# Note that the proxy file is not required and used
+# Note that the proxy file IS required and used in MSUnmerged
 if [ -f /etc/robots/robotkey.pem ]; then
     sudo cp /etc/robots/robotkey.pem $AUTHDIR/dmwm-service-key.pem
     sudo cp /etc/robots/robotcert.pem $AUTHDIR/dmwm-service-cert.pem
@@ -32,6 +86,14 @@
     sudo chmod 0400 $AUTHDIR/dmwm-service-key.pem
 fi
 
+# overwrite proxy if it is present in /etc/proxy
+if [ -f /etc/proxy/proxy ]; then
+    export X509_USER_PROXY=/etc/proxy/proxy
+    mkdir -p /data/srv/state/$srv/proxy
+    [[ -h $AUTHDIR/proxy.cert ]] && rm -rf $AUTHDIR/proxy.cert
+    ln -s /etc/proxy/proxy $AUTHDIR/proxy.cert
+fi
+
 if [ -e $AUTHDIR/dmwm-service-cert.pem ] && [ -e $AUTHDIR/dmwm-service-key.pem ]; then
     export X509_USER_CERT=$AUTHDIR/dmwm-service-cert.pem
     export X509_USER_KEY=$AUTHDIR/dmwm-service-key.pem
@@ -74,6 +136,11 @@
    sudo ln -s /usr/local/data /data/srv/current/apps/reqmgr2
 fi
 
+# Edit the MSUnmerged configuration file (msConfig for the service) at runtime
+# based on the set of arguments passed to the run.sh script by the service
+# statrtup command defined in the .yaml file of the currently starting service
+[[ -n $rseExpr ]] && sed -i -e "s/^[[:blank:]]*RSEEXPR.*/RSEEXPR = \"${rseExpr}\"/g" $CFGFILE
+
 # start the service
 wmc-httpd -r -d $STATEDIR -l "|rotatelogs $LOGDIR/$srv-%Y%m%d-`hostname -s`.log 86400" $CFGFILE

```

### CODE REVIEW

**Comment by todor-ivanov** [comment]
Hi @amaltaro please go ahead and review those changes, so we can deploy this later today

**Comment by amaltaro** [comment]
@todor-ivanov I think you need to make run.sh executable. Please update its permissions in this PR (right now it is 644).

**Comment by todor-ivanov** [comment]
hi @amaltaro  I just changed them
Please take another look

File: docker/pypi/reqmgr2ms-unmerged/run.sh [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+srv=`echo $USER | sed -e "s,_,,g"`
+STATEDIR=/data/srv/state/$srv
+LOGDIR=/data/srv/logs/$srv
+AUTHDIR=/data/srv/current/auth/$srv
+CONFIGDIR=/data/srv/current/config/$srv
+CONFIGFILE=${CONFIGFILE:-config.py}
+CFGFILE=/etc/secrets/$CONFIGFILE
+
+### permission update to workaround issues with mounting logs volume
+sudo chown -R $USER.$USER /data
```
Comment (amaltaro) [inline docker/pypi/reqmgr2ms-unmerged/run.sh:67]: Just an observation for the future, this command is already executed in the Dockerfile and it takes a few minutes to complete. In the future, we need to have a solution that will not delay the startup of the service.

**Review by amaltaro — APPROVED** [review]
Looks good to me and it has now been deployed to production as well. Thanks Todor.

**Pull request merged**